################################################################################
SHORT DESCRIPTION: 
################################################################################
Do not offload ip checksum to guests.

################################################################################
LONG DESCRIPTION: 
################################################################################
Disable NETIF_F_IP_CSUM_BIT & NETIF_F_IPV6_CSUM_BIT for xen-netdev.

################################################################################
CHANGELOG 
################################################################################
Original Author: unknown
Port to 3.18: Eric Chanudet <chanudete@ainfosec.com>
Port to 4.14: Richard Turner <turnerr@ainfosec.com>
Port to 4.19: Richard Turner <turnerr@ainfosec.com>

################################################################################
REMOVAL 
################################################################################
"feature-no-csum-offload" and "feature-ipv6-csum-offload" Xenstore nodes can be
used as well to enable the feature or not, this patch only forces it to off in
any case.

################################################################################
UPSTREAM PLAN
################################################################################
None, could be dealt with at guest creation and be configurable.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
Unknown.

################################################################################
PATCHES 
################################################################################
--- a/drivers/net/xen-netback/interface.c
+++ b/drivers/net/xen-netback/interface.c
@@ -486,9 +486,7 @@ struct xenvif *xenvif_alloc(struct devic
 	INIT_LIST_HEAD(&vif->fe_mcast_addr);
 
 	dev->netdev_ops	= &xenvif_netdev_ops;
-	dev->hw_features = NETIF_F_SG |
-		NETIF_F_IP_CSUM | NETIF_F_IPV6_CSUM |
-		NETIF_F_TSO | NETIF_F_TSO6 | NETIF_F_FRAGLIST;
+	dev->hw_features = NETIF_F_SG | NETIF_F_TSO | NETIF_F_TSO6;
 	dev->features = dev->hw_features | NETIF_F_RXCSUM;
 	dev->ethtool_ops = &xenvif_ethtool_ops;
 
--- a/drivers/net/xen-netfront.c
+++ b/drivers/net/xen-netfront.c
@@ -1314,11 +1314,8 @@ static struct net_device *xennet_create_
 
 	netdev->netdev_ops	= &xennet_netdev_ops;
 
-	netdev->features        = NETIF_F_IP_CSUM | NETIF_F_RXCSUM |
-				  NETIF_F_GSO_ROBUST;
-	netdev->hw_features	= NETIF_F_SG |
-				  NETIF_F_IPV6_CSUM |
-				  NETIF_F_TSO | NETIF_F_TSO6;
+	netdev->features        = NETIF_F_RXCSUM | NETIF_F_GSO_ROBUST;
+	netdev->hw_features	= NETIF_F_SG | NETIF_F_TSO | NETIF_F_TSO6;
 
 	/*
          * Assume that all hw features are available for now. This set
